pkg/gitconfig/config.go

```diff
@@ -0,0 +1,379 @@
+package gitconfig
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/gopasspw/gopass/pkg/debug"
+)
+
+var keyValueTpl = "\t%s = %s%s"
+
+// Config is a single parsed config file. It contains a reference of the input file, if any.
+// It can only be populated only by reading the environment variables.
+type Config struct {
+	path     string
+	readonly bool // do not allow modifying values (even in memory)
+	noWrites bool // do not persist changes to disk (e.g. for tests)
+	raw      strings.Builder
+	vars     map[string]string
+}
+
+// Unset deletes a key.
+func (c *Config) Unset(key string) error {
+	if c.readonly {
+		return nil
+	}
+
+	_, present := c.vars[key]
+	if !present {
+		return nil
+	}
+
+	delete(c.vars, key)
+
+	return c.rewriteRaw(key, "", func(fKey, key, value, comment string) (string, bool) {
+		return "", true
+	})
+}
+
+// IsSet returns true if the key was set in this config.
+func (c *Config) IsSet(key string) bool {
+	_, present := c.vars[key]
+
+	return present
+}
+
+// Set updates or adds a key in the config. If possible it will also update the underlying
+// config file on disk.
+func (c *Config) Set(key, value string) error {
+	section, _, subkey := splitKey(key)
+	if section == "" || subkey == "" {
+		return fmt.Errorf("invalid key: %s", key)
+	}
+
+	// can't set env vars
+	if c.readonly {
+		debug.Log("can not write to a readonly config")
+
+		return nil
+	}
+
+	if c.vars == nil {
+		c.vars = make(map[string]string, 16)
+	}
```

Comment:
This feels like a patch of a symptom of another deeper bug around `c.vars` initialization, no?
Also note that Parse below is initializing with 42 instead of 16. Maybe we should strive to be consistent or have a default magic number as a const?

Reply:
Hopefully my proposed changes above should patch this...

Reply:
I guess actually we can also just keep it `nil` until we need to write to it, true. But then we need to guard against accessing it when it's nil

Reply:
I think we do that.